boards/shared/Inc/App/App_SharedClock.h

```diff
@@ -42,3 +51,11 @@ uint32_t
  * @return The current time for the given clock, in seconds
  */
 uint32_t App_SharedClock_GetCurrentTimeInSeconds(const struct Clock *clock);
+
+/**
+ * Get the previous time for the given clock, in milliseconds
+ * @param clock The clock to get the previous time for
+ * @return The current time for the given clock, in milliseconds
+ */
+uint32_t
+    App_SharedClock_GetPreviousTimeInMilliseconds(const struct Clock *clock);
```

Comment:
Maybe add a previous time in seconds as well? Otherwise LGTM

Reply:
Hmm yeah shore I can add prev time in seconds